llvm/test/CodeGen/X86/tail-merging-preserve-debugloc.ll

```diff
@@ -0,0 +1,31 @@
+; RUN: llc < %s -mtriple=x86_64-unknown-linux-gnu -enable-tail-merge=1 -stop-after=branch-folder | FileCheck %s
```

Comment:
Given that this patch is affecting only branch folding, it makes sense to have the test only cover the branch folding pass -- if you run llc with `-stop-before=branch-folder` you'll get some MIR output, which you can then apply a single pass to by giving llc the `--run-pass=branch-folder`. (Benefits: the test won't be affected by earlier passes, and people reading the test will see the direct input to the branch-folder pass).